### PR TITLE
rewrite converters

### PR DESCRIPTION
**Pull request by NCPlayz**
### Description
Rewrites the converters system to be much more stable and coherent.

### Checklist

- [x] This PR makes changes to the code.
    - [x] The changes are breaking.
    - [x] The changes implement a new feature.
    - [x] The changes fix an issue.
    - [ ] I updated the documentation to reflect these changes.
- [ ] This PR makes changes to the documentation.
    - [ ] The changes require a different version of sphinx.
    - [ ] I updated the configuration file to reflect these changes.

- [x] This PR has been tested.

### CODE REVIEW

File: .gitignore [hunk 1]
```diff
@@ -1 +1,3 @@
 __pycache__/
+pythonenv*/
+build/
```
Comment (ShineyDev) [inline .gitignore]: Unsure where `build/` flew out from, but the `pythonenv*/` entry should exist in your [user-local](https://docs.github.com/en/free-pro-team@latest/github/using-git/ignoring-files#excluding-local-files-without-creating-a-gitignore-file) or [user-global](https://docs.github.com/en/free-pro-team@latest/github/using-git/ignoring-files#configuring-ignored-files-for-all-repositories-on-your-computer) gitignore file.

File: discord/ext/converters/custom_converters/uuid.py [hunk 2]
```diff
@@ -0,0 +1 @@
+# TODO
```
Comment (ShineyDev) [inline discord/ext/converters/custom_converters/uuid.py]: Should this PR be marked as a draft? 🤔

File: discord/ext/converters/model.py [hunk 3]
```diff
@@ -0,0 +1,57 @@
...
+}
+
+
+class ConverterDict(dict):
+    def __init__(self):
+        super().__init__(BASE_CONVERTERS)
+    
+    def __setitem__(self, k: typing.Any, v: typing.Any):
+        if not (issubclass(v, converter.Converter) or inspect.isbuiltin(v)):
+            raise TypeError('Excepted value of type \'Converter\' or built-in, received %r' % v.__name__)
```
Comment (ShineyDev) [inline discord/ext/converters/model.py]:
```suggestion
            raise TypeError("Excepted value of type 'Converter' or built-in, received %r" % v.__name__)
```

Why are you still using Python 2 format syntax?

File: examples/yert.py [hunk 4]
```diff
@@ -0,0 +1,16 @@
+from discord.ext.converters import ConvertersBot
+
+cbot = ConvertersBot(command_prefix='!')
```
Comment (ShineyDev) [inline examples/yert.py]:
```suggestion
bot = ConvertersBot(command_prefix='!')
```

File: examples/yert.py [hunk 5]
```diff
@@ -0,0 +1,16 @@
...
+
+class Yert:
```
Comment (ShineyDev) [inline examples/yert.py:5]: Please create a more realistic example.

**Review by ShineyDev — CHANGES_REQUESTED** [review]

File: discord/ext/converters/core.py [hunk 6]
```diff
@@ -0,0 +1,70 @@
...
+
+    return decorator
+
+def group(name=None, **attrs):
+    """A decorator that transforms a function into a :class:`.Group`.
+    This is similar to the :func:`.command` decorator but the ``cls``
+    parameter is set to :class:`Group` by default.
+    .. versionchanged:: 1.1
+        The ``cls`` parameter can now be passed.
+    """
```
Comment (Gobot1234) [inline discord/ext/converters/core.py]: Can't quite put my finger on where this is from :) https://github.com/Rapptz/discord.py/blob/master/discord/ext/commands/core.py#L1417-L1424

**Pull request merged**